### DISCUSSION TITLE
How to use the glossary in automatic translations?

### DISCUSSION THREAD

**Discussion by amq** [WeblateOrg/weblate#9063] [Q&A]
I would like to have a few terms that are always translated in the same way, no matter what the automatic translation suggests.

Is there a way to do it?

Ideally, weblate should even use the built-in glossary functionality of deepl.

**Comment by nijel** [comment]
Right now, there is no support for synchronizing glossary to machine translation services.

**Comment by amq** [comment]
I would be willing to donate €1k for this functionality, would that be interesting @nijel?

**Reply by nijel** [reply]
Yes, it's definitely the kind of feature that makes sense for Weblate. It just needs to be properly specified first.

The machine translation glossaries will always be limited, while on Weblate there is virtually no limit on size, so that's one of the things to consider. For example, DeepL and Google share similar limits:

DeepL
> The maximum size limit for a glossary is 10 MiB = 10485760 bytes and each source/target text, as well as the name of the glossary, is limited to 1024 UTF-8 bytes. A total of 1000 glossaries are allowed per account.

Google:
> The total number of terms in an glossary input file can't exceed 10.4 million (10,485,760) UTF-8 bytes for all terms in all the languages combined. Any single glossary term must be less than 1024 UTF-8 bytes. Terms longer than 1024 bytes are ignored.
> Maximum number of glossary resources for a project | 10,000


Can you please describe the feature in more detail in the [issue tracker](https://github.com/WeblateOrg/weblate/issues/new?assignees=&labels=&template=feature_request.yml)?

**Reply by nijel** [reply]
Ah, there already is an issue covering this: https://github.com/WeblateOrg/weblate/issues/7086

**Reply by amq** [reply]
Just donated $1k on behalf of ToolSense

**Reply by nijel** [reply]
Thanks, I will try to fit this into our schedule.

**Reply by nijel** [reply]
Sorry, it took longer than I originally anticipated, but it has finally landed: https://github.com/WeblateOrg/weblate/pull/10519
